src/core/engines/Cesium/Feature/Raster/hooks.ts

```diff
@@ -0,0 +1,166 @@
+import { VectorTileFeature } from "@mapbox/vector-tile";
+import { ImageryLayerCollection, ImageryProvider, WebMapServiceImageryProvider } from "cesium";
+import { MVTImageryProvider } from "cesium-mvt-imagery-provider";
+import { useEffect, useMemo, useRef, useState, useReducer } from "react";
+import { useCesium } from "resium";
+
+import { ComputedLayer, Feature } from "@reearth/core/mantle";
+
+import { extractSimpleLayerData } from "../utils";
+
+import { Props } from ".";
+
+const useImageryProvider = (imageryProvider: ImageryProvider | undefined) => {
+  const { viewer } = useCesium();
+  useEffect(() => {
+    if (!imageryProvider) return;
+    const imageryLayers: ImageryLayerCollection = viewer.imageryLayers;
+    const layer = imageryLayers.addImageryProvider(imageryProvider);
+    return () => {
+      imageryLayers.remove(layer);
+    };
+  }, [imageryProvider, viewer]);
+};
+
+const useData = (layer: ComputedLayer | undefined) => {
+  return useMemo(() => {
+    const data = extractSimpleLayerData(layer);
+    return {
+      type: data?.type,
+      url: data?.url,
+      layers: data?.layers
+        ? Array.isArray(data.layers)
+          ? data.layers.join(",")
+          : data?.layers
+        : undefined,
+    };
+  }, [layer]);
+};
+
+export const useWMS = ({
+  isVisible,
+  property,
+  layer,
+}: Pick<Props, "isVisible" | "property" | "layer">) => {
+  const { minimumLevel, maximumLevel, credit } = property ?? {};
+  const { type, url, layers } = useData(layer);
+
+  const imageryProvider = useMemo(() => {
+    if (!isVisible || !url || !layers) return;
+    if (type !== "wms") {
+      return;
+    }
```

Comment:
These can be combined and cleaned up.

Reply:
I fixed it!
[fix: simplify condition](https://github.com/reearth/reearth-web/pull/388/commits/a4633d97584531288c5e3738102c30cd1d11c66b)